### PR TITLE
Refactor: Add `null` to `IFood`

### PR DESCRIPTION
**Pull request by nick-y-ito**
## Overview

Added type `null` to optional properties of` IFood`. Whether or not the property is optional is defined in our backend's [Web API document](https://nishiki-tech.github.io/nishiki-documents/web-api/index.html#tag/container/paths/~1containers~1%7BcontainerId%7D/get).

## Changes

- Added `null` to the `quantity`, `unit` and `expiry` properties of `IFood` in `definition.ts`
- Fix errors caused by adding null

## Notes

-

## Assignee Checklist:

- [x] The naming convention of the PR title is correct (See the comment at the top of this template)
- [x] The base branch is correct (See: [Types of Branches](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [x] The branch name follows the [Branch Naming Conventions](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [x] The correct assignees and reviewers have been designated for this PR
- [x] The coding style follows the [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [x] All the related issues are associated with this PR
- [x] All criteria in the associated issues are met (please tick the checkboxes)
- [x] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

- [x] Double-check the "Assignee Checklist"
- [x] The code follows the generic best practices and our [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [x] The code is well-commented and easy to understand
- [x] The UI changes accurately reflect the provided design

### CODE REVIEW

File: src/types/definition.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ export interface IFood {
   name: string;
   quantity: number;
```
Comment (kanta1207) [inline src/types/definition.ts:47]: Based on the document, is quantity also supposed to be nullable? 
Screenshot 2024-02-07 at 23 15 29
Correct me if I'm missing some context on issue or something 🙏

**Review by kanta1207 — COMMENTED** [review]
Thank you for updating quickly! 
I just left my concern about `quantity` to be nullable or not, so please share your thoughts on that whenever you can!

File: src/types/definition.ts (hunk 1 above)
Comment (nick-y-ito) [reply src/types/definition.ts]: @kanta1207
I apologize for my oversight. I have corrected the mistake, please review it again.

**Review by kanta1207 — APPROVED** [review]
Thank you for the fix and sorry for the delay!
It's looking good to me now!!

**Comment by nick-y-ito** [comment]
@kanta1207 
Thank you for you review :)

**Pull request merged**